src/yadism/coefficient_functions/heavy/kernels.py

```diff
@@ -1,37 +1,53 @@
-# -*- coding: utf-8 -*-
+"""Truely heavy structure function kernels."""
 from eko import basis_rotation as br
 
 from .. import kernels
 from ..light.kernels import nc_weights as light_nc_weights
 
 
 def import_pc_module(kind, process):
+    """Import structure function submodule.
+
+    Parameters
+    ----------
+    kind : str
+        structure functions kind
+    process : str
+        DIS process type
+
+    Returns
+    -------
+    module :
+        suitable submodule
+    """
     return kernels.import_local(kind, process, __name__)
 
 
 def generate(esf, nf, ihq):
     """
-    Collect the heavy coefficient functions
+    Collect the heavy coefficient functions.
 
     Parameters
     ----------
-        esf : EvaluatedStructureFunction
-            kinematic point
-        nf : int
-            number of light flavors
+    esf : EvaluatedStructureFunction
+        kinematic point
+    nf : int
+        number of light flavors
+    ihq : int
+        quark flavor to activate
 
     Returns
     -------
-        elems : list(yadism.kernels.Kernel)
-            list of elements
+    elems : list(yadism.kernels.Kernel)
+        list of elements
+
     """
     kind = esf.info.obs_name.kind
     pcs = import_pc_module(kind, esf.process)
     m2hq = esf.info.m2hq[ihq - 4]
     if esf.process == "CC":
-        # TODO: use ihq for couplings
         w = kernels.cc_weights(
-            esf.info.coupling_constants, esf.Q2, kind, br.quark_names[nf], nf
+            esf.info.coupling_constants, esf.Q2, kind, br.quark_names[ihq - 1], nf
```

Comment:
why do you need to pass `ihq` here?  If you have `F_bottom` in `nf=3` only `g,u,d,s` (and relative - pids) are around right?

Reply:
yes - because that means with the bottom mask ([remember the masks in CC](https://yadism.readthedocs.io/en/latest/theory/fns.html)), we have $V_{ub}$ in game and so `F_bottom` can be non zero (exactly due to $u$-PDF - and of course at NLO through $g$)

Reply:
okay I see thanks for the clarification.